cdap-ui/app/cdap/components/Header/ControlCenterDropdown/index.js

```diff
@@ -108,12 +112,60 @@ export default class ControlCenterDropdown extends Component {
     return false;
   };
 
+  renderEntitiesLink() {
+    const baseCDAPURL = `/ns/${this.props.namespace}`;
+    return (
+      <DropdownItem tag="li">
+        <NavLinkWrapper
+          isNativeLink={this.props.nativeLink}
+          to={this.props.nativeLink ? `/cdap${baseCDAPURL}` : baseCDAPURL}
+          isActive={this.isEntitiesActive}
+        >
+          {T.translate(`${PREFIX}.entities`)}
+        </NavLinkWrapper>
+      </DropdownItem>
+    );
+  }
 
-  render() {
-    let baseCDAPURL = `/ns/${this.props.namespace}`;
-    let dashboardURL = `${baseCDAPURL}/operations`;
-    let reportsURL = `${baseCDAPURL}/reports`;
+  renderDashboardLink() {
+    if (hideDashboard()) {
+      return null;
+    }
+
+    const dashboardURL = `/ns/${this.props.namespace}/operations`;
+    return (
+      <DropdownItem tag="li">
+        <NavLinkWrapper
+          isNativeLink={this.props.nativeLink}
+          to={this.props.nativeLink ? `/cdap${dashboardURL}` : dashboardURL}
+          isActive={this.isDashboardActive}
+        >
+          {T.translate(`${PREFIX}.dashboard`)}
+        </NavLinkWrapper>
+      </DropdownItem>
+    );
+  }
+
+  renderReportsLink() {
+    if (hideReports()) {
+      return null;
+    }
+
+    const reportsURL = `/ns/${this.props.namespace}/reports`;
+    return (
+      <DropdownItem tag="li">
+        <NavLinkWrapper
+          isNativeLink={this.props.nativeLink}
+          to={this.props.nativeLink ? `/cdap${reportsURL}` : reportsURL}
+          isActive={this.isReportsActive}
+        >
+          {T.translate(`${PREFIX}.reports`)}
+        </NavLinkWrapper>
+      </DropdownItem>
+    );
```

Comment:
Would it be better to just separate these `render*` methods to separate components?